internal/services/appservice/function_app_flex_consumption_resource.go

```diff
@@ -420,47 +713,90 @@ func (r FunctionAppFlexConsumptionResource) Create() sdk.ResourceFunc {
 				return fmt.Errorf("the Site Name %q failed the availability check: %+v", id.SiteName, *model.Message)
 			}
 
+			functionAppStorageAccountString := functionAppFlexConsumption.StorageAccountName
+			if !functionAppFlexConsumption.StorageAccountUsesMSI {
+				if functionAppFlexConsumption.StorageAccountKeyVaultSecretID != "" {
+					functionAppStorageAccountString = fmt.Sprintf(helpers.StorageStringFmtKV, functionAppFlexConsumption.StorageAccountKeyVaultSecretID)
+				} else if functionAppStorageAccountString != "" {
+					functionAppStorageAccountString = fmt.Sprintf(helpers.StorageStringFmt, functionAppFlexConsumption.StorageAccountName, functionAppFlexConsumption.StorageAccountAccessKey, *storageDomainSuffix)
+				}
+			}
```

Comment:
This may be more readable / maintainable as a `switch` statement.

Reply:
Thanks @wyattfry for the comment, Would you mind be more specific about which variable to put in the switch condition? This code is calculating the storage string based on the other two properties, just like `azurerm_linux_function_app` did.